### ISSUE TITLE
Way to escape or replace newline characters?

### ISSUE THREAD

**Issue by refactorsaurusrex** [tomasbjerre/pull-request-notifier-for-bitbucket#254]
I use this plugin to trigger HipChat notifications for comments posted to our team's PRs. However, I've discovered that any comment that includes a newline character does not work. Sure enough, if I attempt to send a HipChat notification via http request with something like this:

```
{
  "color": "yellow",
  "message": "blah blah
  blah balh",
  "notify": true,
  "message_format": "html",
  "from": "bob"
}
```

I get the following response:

```
{
    "error": {
        "code": 400,
        "message": "The request body cannot be parsed as valid JSON: Invalid control character u'\\r' at: line 3 column 24 (char 48)",
        "type": "Bad Request"
    }
}
```

That's true regardless of whether the `message_format` is `html` or `text`. Unfortunately, I do not have access to our Bitbucket server logs and I can't seem to find the actual request body in Chrome dev tools (not sure why), so this is all the evidence I currently have to support my theory about what's happening. Assuming I'm correct, are there any known workarounds? Anyway to escape or filter comment text? Using `<br />` for html formats or `\r\n`, `\r`, or `\n` for text formats all work as expected when I use postman to make the hipchat api request.

Thanks for the help!

**Comment by tomasbjerre** [comment]
There is an option "**Post content variables encoding**". Perhaps selecting "**Encode as HTML**" helps.

**Comment by refactorsaurusrex** [comment]
Sorry, I should have included in my original post that I already tried that. No effect.

**Comment by refactorsaurusrex** [comment]
Confirmed: Using `Encode as HTML` has no effect on new line characters for either **html** or **text** messages.

**Comment by tomasbjerre** [comment]
I released a fix in 3.16. Open issue again if not working.

**Comment by refactorsaurusrex** [comment]
Excellent. Thanks!
